modules/gltf/skin_tool.cpp

```diff
@@ -403,11 +403,19 @@ Error SkinTool::_determine_skeletons(
 		Vector<SkinNodeIndex> non_joints;
 		for (int i = 0; i < skeleton_nodes.size(); ++i) {
 			const SkinNodeIndex node_i = skeleton_nodes[i];
-
-			if (nodes[node_i]->joint) {
+			Ref<GLTFNode> node = nodes[node_i];
+			if (node->joint) {
 				skeleton->joints.push_back(node_i);
-			} else {
-				non_joints.push_back(node_i);
+				// If a joint node has non-joint parents, we need to make them joints as well.
```

Comment:
Since we're changing the algorithm for the node tree structure and that breaks the scene tree when doing inherited scenes. We've tended to do a forward upgrade import boolean.

Reply:
It's been a long time since we've had one, but maybe @lyuma remember how we did it.

Reply:
The set of generated nodes with this PR should be a superset of the nodes generated in master.

Reply:
For example inherited nodes will have a different node path. Let's say when we place a magical rod in the bone attachment. So I'm worried this will shift if the Paths are not exactly the same. I can't experiment on this due to a pending trip.